OpenRA.Mods.Common/Scripting/Properties/ProductionProperties.cs

```diff
@@ -244,7 +244,7 @@ public bool Build(string[] actorTypes, LuaFunction actionFunc = null)
 			foreach (var actorType in actorTypes.Distinct())
 				typeToQueueMap.Add(actorType, GetBuildableInfo(actorType).Queue.First());
 
-			var queueTypes = typeToQueueMap.Values.Distinct().ToList();
+			var queueTypes = typeToQueueMap.Values.ToHashSet();
```

Comment:
`queueTypes` lives for a long time and could be iterated many times via `ProductionHandler`. `Distinct().ToArray()` would be the best option here.